lisa/test_runner/tests/test_lisarunner.py

```diff
@@ -0,0 +1,411 @@
+import asyncio
+from typing import Optional
+from unittest.case import TestCase
+
+from lisa import schema
+from lisa.environment import load_environments
+from lisa.test_runner.lisarunner import LisaRunner
+from lisa.tests import test_platform, test_testsuite
+from lisa.tests.test_environment import generate_runbook as generate_env_runbook
+from lisa.tests.test_platform import deleted_envs, deployed_envs, prepared_envs
+from lisa.tests.test_testsuite import (
+    cleanup_cases_metadata,
+    generate_cases_metadata,
+    generate_cases_result,
+)
+from lisa.testsuite import TestStatus, simple_requirement
+from lisa.util import constants
+
+
+def generate_lisarunner(
+    env_runbook: Optional[schema.EnvironmentRoot] = None, case_use_new_env: bool = False
+) -> LisaRunner:
+    runbook = schema.Runbook(
+        platform=[
+            schema.Platform(type=constants.PLATFORM_MOCK, admin_password="not use it")
+        ],
+        testcase=[
+            schema.TestCase(
+                criteria=schema.Criteria(priority=[0, 1, 2]),
+                use_new_environment=case_use_new_env,
+            )
+        ],
+    )
+    if env_runbook:
+        runbook.environment = env_runbook
+    runner = LisaRunner()
+    runner.config(constants.CONFIG_RUNBOOK, runbook)
+
+    return runner
+
+
+class LisaRunnerTestCase(TestCase):
+    def tearDown(self) -> None:
```

Comment:
Just noticed an out of place casing style for this function.

Reply:
It's an override method from built-in unittest.TestCase, cannot change it.